momepy/utils.py

```diff
@@ -55,6 +57,12 @@ def _generate_primal(G, gdf_network, fields, multigraph, oneway_column=None):
     G.graph["approach"] = "primal"
     key = 0
     for row in gdf_network.itertuples():
+
+        if (not (isinstance(row.geometry, LineString))):
+            warnings.warn(
+                message="Geometry is not of type LineString", category=RuntimeWarning
+            )
```

Comment:
This can get quite noisy if you have more non-LineString geometries in you GeoDataFrame. I would suggest calling `gdf_network.geom_type.unique()` before the for loop and checking other geometry types against that. That way we raise only one warning, not one for every geom.

Also, I am not sure if we shouldn't raise an error in this case, rather that a warning. @jGaboardi thoughts?

Reply:
I agree. This way, it will be annoying - printing the warning just once is much better. 
My first idea was to throw an error, too. But then i thought that it may break some applications, which are working, yet. 
I was unsure how the backward compatibility policy is.

Reply:
Fair enough. Better not to be too restrictive. But I'd extend the warning then, saying that the graph will not represent the geodataframe truthfully in this case and people should remove unsupported geometry types.

Reply:
> This can get quite noisy if you have more non-LineString geometries in you GeoDataFrame. I would suggest calling gdf_network.geom_type.unique() before the for loop and checking other geometry types against that. That way we raise only one warning, not one for every geom.
>
>Also, I am not sure if we shouldn't raise an error in this case, rather that a warning. @jGaboardi thoughts?

@Thaulino raises a good point, but throwing a warning every time is a bit overkill. I think your suggestion of single warning is a nice compromise. 

Also, how are we currently handling `MultiLineString` objects?

Reply:
Multi* objects will raise when we try to access coords. Shapely will raise an error there.

Reply:
i implemented the suggested changes